### PR TITLE
[doc] update how_to/perf.md

### PR DESCRIPTION
**Pull request by mli**
@ZihengJiang can you provide some instructions for how to use `profiler`?

### CODE REVIEW

File: example/image-classification/common/util.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import requests
-import os
 import subprocess
+import os
 import errno
 
 def download_file(url, local_fname=None, force_write=False):
+    # requests is not default installed
+    import requests
```
Comment (piiswrong) [inline example/image-classification/common/util.py]: Is it necessary?

Comment (mli) [inline example/image-classification/common/util.py]: requests is used by this function

File: docs/how_to/perf.md [hunk 2]
```diff
@@ -1,7 +1,155 @@
...
+[MKL_README.md](https://github.com/dmlc/mxnet/blob/master/MKL_README.md) for
+details
+
+Also setting the following two environment variables may help:
+- `KMP_AFFINITY=granularity=fine,compact,1,0` if there are two physical CPUs
+- `OMP_NUM_THREADS=vCPUs / 2` in which `vGPUs` is the number of virtual CPUs.
+  For linux we can get it by `cat /proc/cpuinfo  | grep processor | wc -l`
+
+Note that MXNet treats all GPU in a single machine as a single device. So when
+specify `gpu(0)` or `gpu()`, all GPU cores in the machine will be used.
```
Comment (howard0su) [inline docs/how_to/perf.md]: I don't think gpu(0) == gpu() at least it is not true in my testing.

File: docs/how_to/perf.md [hunk 3]
```diff
@@ -1,7 +1,155 @@
...
+|   1 |  98.44 | 24.66 |  27.43 |  11.91 |  38.74 | 14.95 |
+|   2 | 121.67 | 27.07 |  24.78 |  11.61 |  44.27 | 15.62 |
+|   4 | 254.39 | 38.51 |  28.76 |  12.86 |  40.88 | 19.01 |
+|   8 | 237.73 | 36.97 |  25.57 |  12.68 |  43.00 | 16.11 |
+|  16 | 280.82 | 40.00 |  20.85 |  11.77 |  55.00 | 16.93 |
+|  32 | 285.41 | 44.40 |  31.03 |  12.45 |  55.70 | 17.02 |
+
+## Nvidia GPU
+
+`CUDNN` often greatly accelerate performance on Nvidia GPUs, check a recent
```
Comment (howard0su) [inline docs/how_to/perf.md]: I would suggest to mentioned something like ' cuDNN often greatly accelerate performance especially Convolution layer's performance.'

File: docs/how_to/perf.md [hunk 4]
```diff
@@ -1,7 +1,155 @@
...
+CUDNN version is used.
+
+Setting the environment `export MXNET_CUDNN_AUTOTUNE_DEFAULT=1` sometimes also helps.
+
+We show performance results various GPUs including K80 (EC2 p2.2xlarge), M40,
```
Comment (howard0su) [inline docs/how_to/perf.md]: results *of*

File: docs/how_to/perf.md [hunk 5]
```diff
@@ -1,7 +1,155 @@
...
+P100 (DGX-1).
```
Comment (howard0su) [inline docs/how_to/perf.md]: and P100

File: docs/how_to/perf.md [hunk 6]
```diff
@@ -1,7 +1,155 @@
...
+  |   1 | 809.94  | 15.14  | 27.20  |
+  |   2 | 1202.93 | 30.34  | 49.55  |
+  |   4 | 1631.37 | 50.59  | 78.31  |
+  |   8 | 1882.74 | 77.75  | 122.45 |
+  |  16 | 2012.04 | 111.11 | 156.79 |
+  |  32 | 1869.69 | 129.98 | 181.53 |
+
+## Multiple Devices
+
+If more than one GPU or machine is used, MXNet uses `kvstore` to communicate
```
Comment (howard0su) [inline docs/how_to/perf.md]: are used

File: docs/how_to/perf.md (hunk 3 above)
Comment (howard0su) [inline docs/how_to/perf.md]: CUDNN is wrong. it should be cuDNN as how nVidia states it.

**Comment by mli** [comment]
@howard0su thanks for the comments

**Pull request merged**